halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/ui/v1/Ui.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.cli.ui.v1;
+
+/**
+ * A shared colored terminal output class
+ */
+public class Ui {
+  private Color color;
+  private boolean verbose;
+
+  public Ui(boolean colorEnabled, boolean verbose) {
+    this.color = new Color(colorEnabled);
+    this.verbose = verbose;
+  }
+
+  private void reset(String message) {
+    try {
+      System.out.print(message);
+    } catch (Exception e) {
+      throw e;
+    } finally {
+      System.out.println(color.RESET);
+    }
+  }
+
+  public void raw(String message) {
+    reset(message);
+  }
+
+  public void info(String message) {
+    if (verbose) {
+      reset(color.BOLD + color.BLUE + ". " + color.RESET + message);
```

Comment:
At least in this case BOLD is a weight not a color.
Does RED + BLUE make sense? I'm assuming not.
RESET doesnt sound like a Color at all.

If that is the case, then Color and Weight are specializations of some abstract class. I'm calling it Flavor because I cant think of anything here.

Now you have a type "PrintablePhrase" which is a {Flavor, String}
Your output message is a sequence of PrintablePhrase to go back and forth.
To get BOLD + BLUE you have an operator on Flavor that is Flavor = Color + Weight.

This is more API overhead than you have now, but I think clarifies what's happening and offers more opportunities to compose messages since you can have helper functions that produce PrintablePhrases and chain them together. You could even have an implicit reset across phrases and issue a Reset if two adjacent phrases have different Flavors. That probably simplifies use as well by decoupling the phrases and encouraging more composition.

And with all the above, you have a NONE color indicating you are not specifying one. Probably a BLACK too. I'm not sure if NONE means the shell default or continue that which came before. You might need some kind of ANY to indicate leave as (dont reset when appending phrases) as opposed to resetting back to the shell default.